go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/stellar/soroban-tools
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
```

Comment:
Why the heck did the IDE change this? :S

Reply:
no idea, but I noticed that too with go1.21.

Reply:
I think it (maybe wrongly) enforces initializing the toolchain field?

https://tip.golang.org/doc/toolchain

Reply:
More info here https://go.dev/blog/toolchain

Reply:
I don't think it's a good idea to enforce the field

Reply:
OK, I think it may have been caused by https://github.com/stellar/soroban-tools/pull/1030 bumping a dependency using 1.21 which cause Go to update the go.mod with the mentioned version.

See https://github.com/dependabot/dependabot-core/issues/7868 for an occurrence of this.

@tsachiherman can you investigate?

Reply:
OK, I found the culprit:

`github.com/creachadair/mds 1.21`

Reply:
```
go mod why -m github.com/creachadair/mds 
# github.com/creachadair/mds
github.com/stellar/soroban-tools/cmd/soroban-rpc/internal
github.com/creachadair/jrpc2
github.com/creachadair/mds/mlink
```

Reply:
Let's revert https://github.com/stellar/soroban-tools/pull/1030 until this is fixed

Reply:
We have never distinguished this, but that requirement is only meaningful for the Go monorepo because it hosts libraries, the Go SDK, and we need to give folks importing the Go SDK into their own apps time to upgrade. For products, it's reasonable to support only a single version of Go. The soroban-rpc already has a dependence on a single version of Rust so supporting older version of Go seems like a maintenance cost/burden we don't need.